x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/time_range_metadata/many_apm_server_versions.spec.ts

```diff
@@ -0,0 +1,276 @@
+import {
+  TRANSACTION_DURATION_HISTOGRAM,
+  TRANSACTION_DURATION_SUMMARY,
+} from '@kbn/apm-plugin/common/es_fields/apm';
+import { ApmDocumentType } from '@kbn/apm-plugin/common/document_type';
+import { RollupInterval } from '@kbn/apm-plugin/common/rollup';
+import { LatencyAggregationType } from '@kbn/apm-plugin/common/latency_aggregation_types';
+import { Readable } from 'stream';
+import type { ApmApiClient } from '../../../../services/apm_api';
+import type { DeploymentAgnosticFtrProviderContext } from '../../../../ftr_provider_context';
+
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const apmApiClient = getService('apmApi');
+  const synthtrace = getService('synthtrace');
+  const es = getService('es');
+
+  const baseTime = new Date('2023-10-01T00:00:00.000Z').getTime();
+  const startLegacy = moment(baseTime).add(0, 'minutes');
+  const start = moment(baseTime).add(5, 'minutes');
+  const endLegacy = moment(baseTime).add(10, 'minutes');
+  const end = moment(baseTime).add(15, 'minutes');
+
+  describe('Time range metadata when there are multiple APM Server versions', () => {
+    describe('when ingesting traces from APM Server with different versions', () => {
+      let apmSynthtraceEsClient: ApmSynthtraceEsClient;
+
+      before(async () => {
+        apmSynthtraceEsClient = await synthtrace.createApmSynthtraceEsClient();
+        await generateTraceDataForService({
+          serviceName: 'synth-java-legacy',
+          start: startLegacy,
+          end: endLegacy,
+          isLegacy: true,
+          synthtrace: apmSynthtraceEsClient,
+        });
+
+        await generateTraceDataForService({
+          serviceName: 'synth-java',
+          start,
+          end,
+          isLegacy: false,
+          synthtrace: apmSynthtraceEsClient,
+        });
+      });
+
+      after(() => {
+        return apmSynthtraceEsClient.clean();
+      });
+
+      it('ingests transaction metrics with transaction.duration.summary', async () => {
+        const res = await es.search({
+          index: 'metrics-apm*',
+          body: {
+            query: {
+              bool: {
+                filter: [
+                  { exists: { field: TRANSACTION_DURATION_HISTOGRAM } },
+                  { exists: { field: TRANSACTION_DURATION_SUMMARY } },
+                ],
+              },
+            },
+          },
+        });
+
+        // @ts-expect-error
+        expect(res.hits.total.value).to.be(20);
+      });
+
+      it('ingests transaction metrics without transaction.duration.summary', async () => {
+        const res = await es.search({
+          index: 'metrics-apm*',
+          body: {
+            query: {
+              bool: {
+                filter: [{ exists: { field: TRANSACTION_DURATION_HISTOGRAM } }],
+                must_not: [{ exists: { field: TRANSACTION_DURATION_SUMMARY } }],
+              },
+            },
+          },
+        });
```

Comment:
same here: if it is part of env setup and not the actual APM functionality test I suggest moving both calls to `before` hook so it is clearly not a "testing" call

Reply:
same here